### ISSUE TITLE
Fix group notification classification in INSendMessageIntent

### ISSUE THREAD

**Issue by ianalexander** [element-hq/element-x-ios#5699]
#### Context

iOS has a feature where it can automatically announce push notifications as they come in. In order for this to work, [chat apps can classify those push notifications as direct messages or group messages](https://developer.apple.com/documentation/usernotifications/implementing-communication-notifications).

[These ios settings then appear in the notification settings](https://s.yimg.com/ny/api/res/1.2/zSCIg.W4j.UcFwfcZ3Cvkw--/YXBwaWQ9aGlnaGxhbmRlcjt3PTk2MDtoPTc0NztjZj13ZWJw/https://media.zenfs.com/en/how_to_geek_999/1260323eb905ecb2eec0380d6c53ef92) that allow users to enable or disable announcements for each type. You can choose to only announce direct messages, as an example.

#### Problem

With only "Announce direct messages" enabled in the iOS notification settings above, the app will announce every single notification I receive -- including group messages that are not direct messages.

#### Steps to reproduce

1. Put in airpods
2. Go to iOS notification settings for element x, ensure announce notifications is enabled, and only "direct messages" are enabled
3. Receive a message in a room
4. Observe the room message is announced, despite only "direct messages" are enabled for announcements

#### Proposed solution

Today, in [`addCommunicationContext`](https://github.com/element-hq/element-x-ios/blob/d2f1e5caeda79958ef44c2d0b180e716d9587a00/NSE/Sources/NotificationContentBuilder.swift#L278-L285), when a notification belonged to a group room, the recipients array of the `INSendMessageIntent` was being populated with both the sender and the current user (me):

https://github.com/element-hq/element-x-ios/blob/d2f1e5caeda79958ef44c2d0b180e716d9587a00/NSE/Sources/NotificationContentBuilder.swift#L271-L285

[According to Apple's INSendMessageIntent documentation](https://developer.apple.com/documentation/usernotifications/implementing-communication-notifications), the sender parameter defines who sent the message, while the recipients parameter defines the destination users of the message. It says not to include the sender or current user in recipients:

`For group message intents, provide an array that contains the other participants of the message for the recipients parameter. For example, in a group of three participants that includes the current user, the sender of the message and the current user are already represented. The recipients array contains one object — the other recipient. Group conversations can also provide a speakableGroupName if the conversation has a name.`

I believe when handling an incoming group notification, including the sender inside the recipients destination array causes some kind of issue. I think iOS interprets a conversation containing only the sender and the receiver (me) as a 1:1 direct message, which overrides the group layout rules and ignores the speakableGroupName.

I believe the right way to fix this is to simply set recipients to `nil` and keep the `speakableGroupName` value:

```
let intent = INSendMessageIntent(recipients: nil,
                                         outgoingMessageType: .outgoingMessageText,
                                         content: nil,
                                         speakableGroupName: speakableGroupName,
                                         conversationIdentifier: notificationContent.roomID,
                                         serviceName: nil,
                                         sender: sender,
                                         attachments: nil)
```

I don't have an ios dev license to check this myself but based on the docs I've read I think this is right.

#### What did you expect?

I expect with only "direct messages" enabled for announcement, room messages are not announced.

#### What happened instead?

With only "direct messages" enabled for announcement, every notification is announced, including room notifications.

### Will you send logs?

No

**Comment by Velin92** [comment]
Thanks for the very in-depth report of the issue, I'll be sure to give a try to your suggested solution, and if it keeps the existing behaviour (which is used to make the notifications have the correct avatar, group name and sender name just like other messaging apps) without impacting notification announcements.

**Comment by Velin92** [comment]
Sadly I have tested the suggested fix, but it utterly breaks the presentation of the notification... I will test some other stuff, since at this point the issue might not be related to this code directly

**Comment by stingalleman** [comment]
Does this also allow you to group notifications? I'm going absolutely crazy receiving a separate notification that buzzes my phone for EVERY notification

**Comment by ianalexander** [comment]
No.

@Velin92 since my proposed solution didn't work I'll defer to you if you wish to close this issue or keep it open.
